### PR TITLE
fix(docs): update workflow API example to include query parameter

### LINKED ISSUE #9628: Docs (Code Issue): Missing Query Parameter in Example of Workflow
### What Medusa version and documentation are you using?

v2

### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/medusajs/medusa/issues

### Issue Summary

In the [documentation](https://docs.medusajs.com/v2/basics/workflows#4-test-workflow), the example provided for making a request to the workflow endpoint is missing the `name` query parameter.

### Actual example:
```
curl http://localhost:9000/workflow
```

### Expected example:
```
curl http://localhost:9000/workflow?name=john
```

This is causing an issue when using the workflow in code, as demonstrated below.

## Code Example:

- Workflow : `src/workflows/hello-world.ts`

```bash
import {
  createStep,
  createWorkflow, StepResponse, WorkflowResponse
} from "@medusajs/framework/workflows-sdk"

// step-1
const step1 = createStep("step-1", async () => {
  return new StepResponse(`Hello from step one!`)
})

// step-2
type WorkflowInput = {
  name: string
}

const step2 = createStep("step-2", async ({ name }: WorkflowInput) => {
  return new StepResponse(`Hello ${name} from step two!`)
})

// workflow
const myWorkflow = createWorkflow(
  "hello-world",
  function (input: WorkflowInput) {
    const str1 = step1()
    // to pass input
    const str2 = step2(input)

    return new WorkflowResponse({
      message: str1,
    })
  }
)

export default myWorkflow


```

- API Route : `src/api/workflow/route.ts`

```bash
import type {
  MedusaRequest,
  MedusaResponse,
} from "@medusajs/framework/http"
import myWorkflow from "../../workflows/hello-world"

export async function GET(
  req: MedusaRequest,
  res: MedusaResponse
) {
  const { transaction, result, errors } = await myWorkflow(req.scope)
    .run({
      input: {
        name: req.query.name as string,
      },
    })

  res.json({
    transaction,
    result,
    errors,
  })
}
```

When running this code with `curl http://localhost:9000/workflow`. I encountered the following issue:

- Console output for a `transaction`: 

```bash
{
  "transaction": {
    # ... other fields
    "context": {
      "payload": {},
      "invoke": {
        "step-1": {
          "__type": "Symbol(WorkflowWorkflowData)",
          "output": {
            "__type": "Symbol(WorkflowStepResponse)",
            "output": "Hello from step one!",
            "compensateInput": "Hello from step one!"
          }
        },
        "step-2": {
          "__type": "Symbol(WorkflowWorkflowData)",
          "output": {
            "__type": "Symbol(WorkflowStepResponse)",
            "output": "Hello undefined from step two!", # undefined
            "compensateInput": "Hello undefined from step two!"  # undefined
          }
        }
      },
      "compensate": {}
    },
   # ... other fields
  },
  "result": {
    "message": "Hello from step one!"
  },
  "errors": []
}

```

The issue arises because the `name` parameter is not passed in the request. Updating the curl command in the [documentation](https://docs.medusajs.com/v2/basics/workflows#4-test-workflow) to include the correct query parameter would resolve this issue.

#### Steps to Reproduce:

1. Implement the code as shown above.
2. Run the curl command `curl http://localhost:9000/workflow` (as per the docs).
3. The output will be `Hello undefined from step two!`

#### Resolution Suggestion: 

Update the [documentation](https://docs.medusajs.com/v2/basics/workflows#4-test-workflow) to include the query parameter in the curl command:

```
curl http://localhost:9000/workflow?name=john
```

### Are you interested in working on this issue?

- [x] I would like to fix this issue

### PR DESCRIPTION
**Pull request by VrajPatelK**
#### Summary
This pull request addresses a missing query parameter in the Medusa documentation's example for making a request to the workflow endpoint.

#### Issue
The example in the documentation incorrectly shows:
```
curl http://localhost:9000/workflow
```
This causes an issue when running the workflow in code, specifically in `step-2`, where the `name` parameter is `undefined`. 

#### Expected Behavior
The correct curl command should include the `name` query parameter, like so:
```
curl http://localhost:9000/workflow?name=john
```

#### Code Example
Here is the code that demonstrates the issue:

```ts
const step2 = createStep("step-2", async ({ name }: WorkflowInput) => {
  return new StepResponse(`Hello ${name} from step two!`)
})
```

When running the incorrect curl command, the output for `step-2` returns:
```
"Hello undefined from step two!"
```

#### Fix
The pull request proposes updating the example curl command in the documentation to:
```
curl http://localhost:9000/workflow?name=john
```

This ensures that the `name` parameter is correctly passed and resolves the issue of `undefined` values in `step-2`.

#### Steps to Reproduce
1. Implement the workflow and API code as described in the [documentation](https://docs.medusajs.com/v2/basics/workflows#4-test-workflow).
2. Run the curl command:
   ```
   curl http://localhost:9000/workflow
   ```
3. The console will output `Hello undefined from step two!` due to the missing `name` parameter.

#### Suggested Resolution
Update the curl command in the [documentation](https://docs.medusajs.com/v2/basics/workflows#4-test-workflow) to:
```
curl http://localhost:9000/workflow?name=john
```

### Related Issue
Link to the issue: [Medusa Issue #9628](https://github.com/medusajs/medusa/issues/9628)

### CODE REVIEW

File: .changeset/wicked-experts-melt.md
Comment (shahednasser) [inline .changeset/wicked-experts-melt.md:1]: Please remove this file as it's not necessary for documentation contributions.

**Review by shahednasser — COMMENTED** [review]
Thanks for the contribution! Just a small change required.

As for the vercel authentication, since the change is small there's no need to deploy it.

**Review by shahednasser — CHANGES_REQUESTED** [review]
@VrajPatelK please remove the file as instructed above

File: .changeset/wicked-experts-melt.md
Comment (VrajPatelK) [reply .changeset/wicked-experts-melt.md:1]: removed please, review it !

**Review by shahednasser — APPROVED** [review]
LGTM thanks!

**Pull request merged**